Zend/zend_strtod.c

```diff
@@ -3613,13 +3613,22 @@ rv_alloc(i) int i;
 rv_alloc(int i)
 #endif
 {
+
 	int k, *r;
+	size_t j, rem;
+
+	rem = sizeof(Bigint) - sizeof(ULong) - sizeof(int);
 
-	size_t j = sizeof(ULong);
+
+	j = sizeof(ULong);
+	if (i > (INT_MAX - rem))
```

Comment:
I think this line should be
```suggestion
	if (i > ((INT_MAX >> 2) + rem))
```
Then the other overflow check should be superfluous.

Reply:
Oh, and I think reverting the changes from #15715 could also be appropriate, since `j` should never exceed `INT_MAX` (actually, it must always be less than `0x40000000`).